src/plugin/modules/widgets/reads_set.js

```diff
@@ -0,0 +1,228 @@
+                        readsSetInfo.set = {
+                            name: results.info[1],
+                            type: results.data.items[0].info[2].split('-')[0].split('.')[1],
+                            description: results.data.description,
+                            items: results.data.items,
+                            item_count: results.data.items.length
+                        };
+                        
+                        // pull all reads objects to calculate summary stats
+                        // and individual browse row contents
+                        for (i = 0; i < readsSetInfo.set.items.length; i++) {
+                            if (reads_type !== readsSetInfo.set.items[i].info[2]) {
+                                readsSetInfo.set.type = "Mixed Reads";
+                            }
+                            refs.push({'ref':readsSetInfo.set.items[i].ref});
+                        }
+
+                        return refs;
+                    })
+                    .then(function (refs) {
+                        return Promise.resolve(wsClient.get_objects(refs));
+                    })
+                    .then(function (data) {
+                        var i = 0,
+                            results = {};
+                        readsSetInfo.row_contents = data;
+                        readsSetInfo.set.stats = {
+                            read_count: 0,
+                            bp_count: 0
+                        };
+                        // calculate summary stats
+                        for (i = 0; i < data.length; i++) {
+                            readsSetInfo.row_contents[i].name = data[i].info[1];
+                            readsSetInfo.row_contents[i].type = data[i].info[2].split('-')[0].split('.')[1];
+                            if (data[i].data.hasOwnProperty('read_count')) {
+                                readsSetInfo.set.stats.read_count += data[i].data.read_count;
+                                if (data[i].data.hasOwnProperty('read_size')) {
+                                    readsSetInfo.set.stats.bp_count += data[i].data.read_count * data[i].data.read_size;
+                                }
+                            }
+                        }
+                        return readsSetInfo;
+                    })
+                    .error(function (err) {
+                        console.error(err);
+                        container.empty();
+                        container.appendChild('Error' + JSON.stringify(err));
+                    });
+            }
+            
+            // WIDGET API
+            function attach(node) {
+                parent = node;
+                container = parent.appendChild(document.createElement('div'));
+                container.innerHTML = layout();
+            }
+
+            function start(params) {
+                Array.from(container.querySelectorAll("[data-element]"))
```

Comment:
Array.from may not be supported on IE. I use MDN compatibility tables when in doubt, but they may not always be up to date: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from
When wanting array access to querySelectorAll one can also use Array.prototype.slice.call.